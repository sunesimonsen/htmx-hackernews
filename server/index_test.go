package server

import (
	"testing"
)

func TestIndex(t *testing.T) {
	t.Run("GET / with a succesful response", func(t *testing.T) {
		snapshotRequest(t, "/", "/v0/topstories.json", []int{42, 3545, 345, 1})
	})
}
