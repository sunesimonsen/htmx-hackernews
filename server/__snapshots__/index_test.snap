
[TestIndex/GET_/_with_a_succesful_response - 1]
http.Header{
    "Cache-Control": {"max-age=0"},
    "Content-Type":  {"text/html"},
    "Etag":          {"W/\"20-9104c9716e266b8757161eaaeeeb3efa4c74525d\""},
}
int(200)
<title>
 Hackernews
</title>
<meta charset="UTF-8">
<meta name="description" content="Hackernews example app in htmx and Go" lang="en">
<link rel="stylesheet" href="/assets/main.css">
<link rel="icon" href="/assets/favicon.ico" type="image/x-icon">
<script src="https://unpkg.com/htmx.org@1.9.4" integrity="sha384-zUfuhFKKZCbHTY6aRR46gxiqszMk5tcHjsVFxnUo8VMus4kHGVdIYVbOYYNlKmHV" crossorigin="anonymous">
</script>
<script src="https://unpkg.com/ulid@2.3.0/dist/index.umd.js">
</script>
<script src="/assets/scripts.js">
</script>
<main>
 <header class="topBar">
  <a class="logo" href="/">
   <img src="/assets/y18.gif">
   <span class="logo-caption">
    Hacker News
   </span>
  </a>
 </header>
 <section class="content">
  <div class="stories">
   <article class="story" hx-get="/parts/story/42" hx-trigger="revealed" hx-swap="outerHTML">
    <a class="title">
     <div class="skeleton" style="width: 60%">
     </div>
    </a>
    <footer>
     <div class="skeleton" style="width: 40%">
     </div>
    </footer>
   </article>
   <article class="story" hx-get="/parts/story/3545" hx-trigger="revealed" hx-swap="outerHTML">
    <a class="title">
     <div class="skeleton" style="width: 60%">
     </div>
    </a>
    <footer>
     <div class="skeleton" style="width: 40%">
     </div>
    </footer>
   </article>
   <article class="story" hx-get="/parts/story/345" hx-trigger="revealed" hx-swap="outerHTML">
    <a class="title">
     <div class="skeleton" style="width: 60%">
     </div>
    </a>
    <footer>
     <div class="skeleton" style="width: 40%">
     </div>
    </footer>
   </article>
   <article class="story" hx-get="/parts/story/1" hx-trigger="revealed" hx-swap="outerHTML">
    <a class="title">
     <div class="skeleton" style="width: 60%">
     </div>
    </a>
    <footer>
     <div class="skeleton" style="width: 40%">
     </div>
    </footer>
   </article>
  </div>
 </section>
</main>

---
